pkg/validation.go

```diff
@@ -147,6 +156,11 @@ func (*Validation) externalLink(link Link) (Link, error) {
 
 			resp.Body.Close()
 			break
+		} else if resp.StatusCode == http.StatusTooManyRequests {
+			status = false
+			message = "Too many requests"
+			v.limiter.Limit()
+			continue
```

Comment:
You forgot about closing the response body. Please refactor the code to use `defer` right after opening the body instead.